### ISSUE TITLE
More control options over the statusline and tabline

### ISSUE THREAD

**Issue by iasj** [neovim/neovim#15432]
### Feature already in Vim?

I don't think so.

### Feature description

Hi there.

I've been following the issues segment for quite a while, and even posted a few requests myself: [here](https://github.com/neovim/neovim/issues/11924#issuecomment-886088202), [here](https://github.com/neovim/neovim/issues/9342#issuecomment-894888992), and [here](https://github.com/neovim/neovim/issues/1435#issuecomment-670837497).

The thing is, I'm building a collection of plug-ins and since they are growing in number, I fear I won't be able to move forward with some ideas because of some limitations.

# Limitation number 1 (global tabline)

This limitation only affects the looks if one the my plugins, but it may be important in the future for other ones. The plugin is call [Zoom](https://gitlab.com/nvpm/zoom) and it's nothing more than a distraction-free implementation. The thing is, Zoom gives support to another plugin now call [nvpm](https://gitlab.com/nvpm/nvpm), which is gonna be broken down into more plugins soon. 

The problem itself is regarding the fact that the tabline is unique to all buffers. The "problem" can be seen in the following screenshot:

![zoom](https://user-images.githubusercontent.com/7888397/130158793-09723513-77de-4e2b-a040-ff5e5744abf0.png)

You can see that the tabline occupies the whole width of the screen, whereas the statusline is only local to the buffer in the middle. 

Solution I want: I want to be able to set the tabline to apear only on the focuses buffer (the one in the middle in the shot above). Moreover, the command-line is also global, and it would be perfect to have it occupy only the width of the focused buffer, right below the statusline.

**Comment by justinmk** [comment]
> Solution I want: to have mouse support for MouseClick, MouseScroll, and MouseHover (if possible) for both tab and status lines.

Nvim tabline supports mouse clicks: https://github.com/neovim/neovim/pull/3871 , via the `%@Func@` format.

`:help 'statusline'`  :
```
	@ N   For 'tabline': start of execute function label. Use %X or %T to 
	      end the label, e.g.: %10@SwitchBuffer@foo.c%X.  Clicking this 
	      label runs specified function: in the example when clicking once 
	      using left mouse button on "foo.c" "SwitchBuffer(10, 1, 'l', 
	      '    ')" expression will be run.  Function receives the 
```

**Comment by iasj** [comment]
>You found #11924 so I guess you know already :)

Yeah I found. Since it still doesn't have support for the statusline as well, I'm afraid the use case I intend would be incomplete, so I won't even bother to implement it for now.

>If you want full control over the statusline, the current answer is to implement a custom UI.

I was thinking the same thing, but I was in denial. I couldn't believe I was gonna have to do this. But yeah, good point.

>Proposal
Instead of adding more features to 'statusline', it could optionally be driven by a user-defined floating window/buffer.

Yeah, I heard about those floating window/buffers somewhere, but I never really found it in the docs. Do you know where?

By the way - making use of this message - here is a shot of what I meant for the local tabline:

![zoom2](https://user-images.githubusercontent.com/7888397/130173693-f344810c-2310-445d-ac19-c8f81428b3df.png)

I'm faking a zoom mode with tmux panes around the pane in the middle. That's the same strategy I use in the [zoom](https://gitlab.com/nvpm/zoom) plug-in, but with Vim buffers instead of tmux panes.

**Comment by iasj** [comment]
>Yeah, I heard about those floating window/buffers somewhere, but I never really found it in the docs. Do you know where?

I think I found a [great tutorial on the matter](https://www.statox.fr/posts/2021/03/breaking_habits_floating_window/). No wonder I haven't found in the docs, for I was looking for tutorials using the string `:help float<tab>`. I'll definitely look into this.

**Comment by danielo515** [comment]
>Instead of adding more features to 'statusline', it could optionally be driven by a user-defined floating window/buffer.

Doesn't floating windows obscure part of the buffer/window you have focused? Even if it just one line, it can get messy if you can not get to the bottom of your file

**Comment by bfredl** [comment]
@danielo515 yes, we would need a new positioning mode to properly allocate space from the parent window with no occlusion

**Comment by danielo515** [comment]
> @danielo515 yes, we would need a new positioning mode to properly allocate space from the parent window with no occlusion

That sounds very interesting. If you allow that, that will open the possibility to have "metadata/child" windows in any position right? That may be awesome. Will those windows be attached to the parent one? Meaning I can move the parent window and they will move with it,or close the parent window and that will close the attached windows?
